clients/Android/ultrablue/app/src/main/java/fr/gouv/ssi/ultrablue/model/UltrablueProtocol.kt

```diff
@@ -109,24 +123,78 @@ class UltrablueProtocol(private val activity: MainActivity, private var device:
             }
             ATTESTATION_SEND_NONCE -> {
                 rand.nextBytes(attestationNonce)
-                logger?.push(CLog("Generated anit replay nonce", true))
+                logger?.push(CLog("Generated anti replay nonce", true))
                 val encoded = Cbor.encodeToByteArray(ByteArrayModel(attestationNonce))
                 writeMsg("anti replay nonce", encoded)
             }
             ATTESTATION_READ -> readMsg("Attestation data")
-            ATTESTATION_PERFORM -> {
+            VERIFY_QUOTE -> {
+                encodedPlatformParameters = message
                 logger?.push(Log("Verifying quotes signature"))
                 try {
-                    Gomobile.checkQuotesSignature(message, encodedAttestationKey, attestationNonce)
+                    Gomobile.checkQuotesSignature(encodedPlatformParameters, encodedAttestationKey, attestationNonce)
+                    logger?.push(CLog("Quotes signature are valid", true))
+                    state += 1
+                    resume()
+                } catch (e: Exception) {
+                    logger?.push(CLog("Error while verifying quote(s) signature: ${e.message}", false))
+                }
+            }
+            REPLAY_EVENT_LOG -> {
+                logger?.push(Log("Replaying event log"))
+                try {
+                    Gomobile.replayEventLog(encodedPlatformParameters)
+                    logger?.push(CLog("Event log has been replayed", true))
+                    state += 1
+                    resume()
                 } catch (e: Exception) {
                     logger?.push(CLog("${e.message}", false))
+                }
+            }
+            PCRS_HANDLE -> {
+                try {
+                    logger?.push(Log("Getting PCRs"))
+                    encodedPCRs = Gomobile.getPCRs(encodedPlatformParameters).data
+                    logger?.push(CLog("Got PCRs", true))
+                } catch (e: Exception) {
+                    logger?.push(CLog("Error while getting PCRs: ${e.message}", false))
                     return
                 }
-                logger?.push(CLog("Quotes signature are valid", true))
+                val response: AttestationResult = if (enroll) {
+                    logger?.push(Log("Storing new attester entry"))
+                    registerDevice()
+                    AttestationResult(false, "")
+                } else {
+                    logger?.push(Log("Comparing PCRs"))
+                    if (device.encodedPCRs.contentEquals(encodedPCRs)) {
+                        logger?.push(CLog("PCRs entries match the stored ones", true))
+                        AttestationResult(false, "")
+                    } else {
+                        // TODO: Need deeper investigation to determine which PCR changed and why.
```

Comment:
A common pratice (at Google anyway) is to create a github issue when you have long-term TODOs to implement, and to reference it in the comment so that you can store more details in the issue and update it later:
// TODO(#42): ...